### PR TITLE
QA Testing Style Guide Update

### PR DESCRIPTION
**Pull request by DonaldJennings**
Added section containing style guide conventions for testing
Changes include 
- Test names
- Test fixture conventions
- Where to declare variables in tests
- Error Messages.

Also added in some comments to separate the sections of the HTML to make it a bit easier in case anyone else comes around one day to change it and doesn't spend ages trying to find where they are in relation to the style guide website.

### CODE REVIEW

**Review by MeinaLinWei — APPROVED** [review]
Look good to me.

File: index.html [hunk 1]
```diff
@@ -522,9 +533,15 @@ <h3 id="Exceptions_to_Naming_Rules" style="left: -46px; position: relative;">Exc
...
       </div>
 
+<!-------------------------------------------------------------------------------------------------------------------
+  Comments
+-------------------------------------------------------------------------------------------------------------------->
       <h2 id="Comments" style="left: -46px; position: relative;">Comments</h2>
 
-      <p>Code should be documented well. This serves several purposes. Well documented code is easier for others to understand, which leads to better code maintainability. Furthermore, once a programmer needs to document code, they need to think about the code structure, design, and functionality. This thinking fosters writing better code in general. Documentation can also be used to generate html site for easy navigation through project’s APIs.</p>
+      <p>Code should be documented well. This serves several purposes. Well documented code is easier for others to understand, 
+         which leads to better code maintainability. Furthermore, once a programmer needs to document code, they need to think
```
Comment (mifrandir) [inline index.html:540]: I like this.

File: index.html [hunk 2]
```diff
@@ -290,7 +298,10 @@ <h3 id="File_Names" style="left: -46px; position: relative;">File Names</h3>
           <code>foo_bar.hpp</code> and
           <code>foo_bar.cpp</code>, defining a class called
           <code>FooBar</code>.</p>
-
+          
+          <p>There is an exemption to this rule when writing tests. In this case tests should have .test.cpp as
```
Comment (mifrandir) [inline index.html]: Why not write "This rule may not apply in testing. For more information see `<link here>`"? And then add the file naming to the test section? I think that would be a bit more concise.

**Review by mifrandir — CHANGES_REQUESTED** [review]

File: index.html (hunk 2 above)
Comment (DonaldJennings) [reply index.html]: Yes that's a good suggestion 😄

**Review by pmorande27 — APPROVED** [review]

**Review by mifrandir — APPROVED** [review]

File: index.html [hunk 3]
```diff
@@ -1799,7 +1817,126 @@ <h3 id="Vertical_Whitespace" style="left: -46px; position: relative;">Vertical W
...
+struct DifferentiatorFunctionality : public ::testing::Test {
+...
+}
+    </pre>
+  </div>
+  
+  <!-- Test Fixture Variables vs Local Test Variable -->
+  <h3 id="Test Fixture Variables vs Local Test Variables" style="left: -46px; position: relative">Test fixture variables vs local test variables</h3>
+  <div class="stylebody">
+    <p> When declaring variables in tests we need to be consistent about where we accessing them from and their scopes. <em><b>All variables which you will use
```
Comment (suryanshmanocha) [inline index.html]: Should it read "we need to be consistent about where we **are** accessing them from and their scopes." ?

Comment (DonaldJennings) [reply index.html]: Yes just noticed that formatting error on my end. It will display fine on the actual page though

Comment (suryanshmanocha) [reply index.html]: Sorry, what do you mean?

Comment (DonaldJennings) [reply index.html]: nevermind ignore what i said, i was reading it on my phone and the formatting for  the <p> got all messed up due to screen size. Was half asleep at the time haha

**Review by suryanshmanocha — APPROVED** [review]
Looks good to me! I'd love to be involved in future PR's too, good stuff!

**Pull request merged**